zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/user/UserCreateProcessor.java

```diff
@@ -118,14 +118,14 @@ final var record = command.getValue();
   private void addUserPermissions(final long key, final String username) {
     final var authorizationRecord =
         new AuthorizationRecord()
-            .setAuthorizationKey(key)
+            .setAuthorizationKey(keyGenerator.nextKey())
             .setOwnerKey(key)
             .setOwnerId(username)
             .setOwnerType(AuthorizationOwnerType.USER)
             .setResourceType(AuthorizationResourceType.USER)
             .setResourceId(username)
             .setAuthorizationPermissions(Set.of(PermissionType.READ, PermissionType.UPDATE));
 
-    commandWriter.appendFollowUpCommand(key, AuthorizationIntent.CREATE, authorizationRecord);
+    commandWriter.appendFollowUpCommand(key, AuthorizationIntent.CREATED, authorizationRecord);
```

Comment:
@nicpuppa I just noticed you write an event using the command writer. Not sure what happens if you do this, but I doubt it'll work 😄

Reply:
Why ?